.circleci/config.yml

```diff
@@ -294,29 +265,49 @@ jobs:
             --build-arg APP_GIT_COMMIT=${CIRCLE_SHA1} \
             --build-arg APP_BRANCH_NAME=${CIRCLE_BRANCH} \
 
-  deploy-dev:
-    executor: cloud-platform-executor
-    steps:
-      - deploy-to-environment:
-          environment: dev
-
-  deploy-dev-crm4:
-    executor: cloud-platform-executor
-    steps:
-      - deploy-to-environment:
-          environment: dev-crm4
-
-  deploy-uat:
+  deploy-to-environment:
+    description: Deploy image to the specified environment
     executor: cloud-platform-executor
+    parameters:
+      environment:
+        description: Name of environment to deploy to
+        type: string
+      registry:
+        description: Location of ECR images
+        type: string
+        default: ${AWS_ECR_REGISTRY_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com
+      dynamic_hostname:
+        type: boolean
     steps:
-      - deploy-to-environment:
-          environment: uat
+      - aws-cli/setup:
+          role_arn: $ECR_ROLE_TO_ASSUME
+          region: $ECR_REGION
+      - checkout
+      - run:
+          name: Authenticate K8s and export environment variables
+          command: |
+            .circleci/authenticate_with_kubernetes_cluster
+            source .circleci/define_build_environment_variables << parameters.dynamic_hostname >>
+            echo "export IMAGE_TAG=$IMAGE_TAG" >> $BASH_ENV
+            echo "export RELEASE_HOST=$RELEASE_HOST" >> $BASH_ENV
+            echo "export RELEASE_NAME=$RELEASE_NAME" >> $BASH_ENV
+      - run:
+          name: Deploy to << parameters.environment >>
+          command: |
+            ./bin/deploy << parameters.registry >> << parameters.environment >>
 
-  deploy-prod:
+  cleanup_merged:
     executor: cloud-platform-executor
     steps:
-      - deploy-to-environment:
-          environment: prod
+      - checkout
+      - run:
+          name: Authenticate with cluster
+          command: |
+            .circleci/authenticate_with_kubernetes_cluster
+      - run:
+          name: Delete dynamic release
```

Comment:
The difficulty with trying to trigger this in CI is that [CircleCI doesn't have a trigger for branch delete or PR close events](https://discuss.circleci.com/t/github-pr-close-or-branch-delete-events/27062), meaning that is a PR is closed without merging it, its dynamic cloud platform environment won't be deleted. I _think_ this means we need to be triggering this via a GitHub action.

Reply:
@patrick-laa makes sense - I think we could go with the github action.